PerformanceTest_Static_PressureDorsal.py

```diff
@@ -0,0 +1,186 @@
+class avgData:
+    avgDorsal: np.array
+    avgPlantar: np.array
+    config: str
+    subject: str
+    fullDat: pd.DataFrame #entire stored dataframe. 
+    #this class is useful for plotting and subsequent analysis
+    
+    # below is a method of the dataclass
+    def plotAvgPressure(self):
+        fig, (ax1, ax2) = plt.subplots(1,2)
+        ax1 = sns.heatmap(self.avgDorsal, ax = ax1, cmap="mako", vmin = 0, vmax = np.max(self.avgDorsal) * 2)
+        ax1.set(xticklabels=[])
+        ax1.set_title('Shin Pressure') 
+        ax2 = sns.heatmap(self.avgPlantar, ax = ax2, cmap="mako", vmin = 0, vmax = np.max(self.avgPlantar) * 2)
+        ax2.set(xticklabels=[])
+        ax2.set_title('Calf Pressure') 
+        plt.suptitle(self.config)
+        plt.tight_layout()
+        return fig  
+    
+    def sortDF(self, colName):
+        """ 
+        Grabs each individual grouping by location of foot from regions 
+        specified in XSENSOR output
+        """
+        subsetDat = self.fullDat.iloc[:,self.fullDat.columns.get_loc(colName):self.fullDat.columns.get_loc(colName)+12]
+        return(subsetDat)
+
+    
+
+def createAvgMat(inputName):
+    """ 
+    Reads in file, creates average matrix data to be plotted and features
+    are extracted. The result is a dataclass which can be used for further plotting
+    """
+    dat = pd.read_csv(fPath+inputName, sep=',', skiprows = 1, header = 'infer')
+    subj = inputName.split(sep="_")[0]
+    config = inputName.split(sep="_")[1].split(sep=".")[0]
+    sensel = dat.iloc[:,17:197]
+    plantarSensel = dat.iloc[:,214:425]
+    
+    headers = plantarSensel.columns
+    store_r = []
+    store_c = []
+
+    for name in headers:
+        store_r.append(int(name.split(sep = "_")[1])-1)
+        store_c.append(int(name.split(sep = "_")[2].split(sep=".")[0])-1)
+    
+    con_press = np.zeros((np.max(store_r)+1,np.max(store_c)+1))
+    
+    for ii in range(len(headers)-1):
+        con_press[store_r[ii],store_c[ii]] = np.mean(plantarSensel.iloc[:,ii])
+        
+   
+    avgDorsalMat = np.array(np.mean(sensel, axis = 0)).reshape((18,10))
+    avgPlantarMat = np.array(np.flip(con_press))
+    
+    result = avgData(avgDorsalMat, avgPlantarMat, config, subj, dat)
+    
+    return(result)
+
+
+meanDorsalPressure = []
+maxDorsalPressure = [] 
+sdDorsalPressure = []
+totalDorsalPressure = []
+config = []
+subject = []
+
+plantarContact = []
+plantarPeakPressure = []
+plantarAvgPressure = []
+plantarSDPressure = []
+plantarTotalPressure = []
+
+for entry in entries:
+    
+    entry = entries[0]
+    tmpAvgMat = createAvgMat(entry)
+    tmpAvgMat.plotAvgPressure()
+    answer = messagebox.askyesno("Question","Is data clean?")
+    
+    plantar = tmpAvgMat.avgPlantar
+    
+    if answer == False:
+        plt.close('all')
+        print('Adding file to bad file list')
+        #badFileList.append(fName)
+    
+    if answer == True:
+        plt.close('all')
+        print('Estimating point estimates')
```

Comment:
Probably want to remove this plotting for the final version. I was getting slightly confused when the previous trial was plotting during the debug plot.